### PR TITLE
Allow fixed to false positive transition

### PR DESCRIPTION
**Pull request by jesusfcr**

### CODE REVIEW

File: pkg/api/service/finding.go [hunk 1]
```diff
@@ -72,12 +65,33 @@ func (s vulcanitoService) ListFindingOverwrites(ctx context.Context, findingID s
...
+		return true
+	}
+
+	validTransitions := map[string]any{
+		"OPEN->FALSE_POSITIVE":  nil,
+		"FALSE_POSITIVE->OPEN":  nil,
+		"FIXED->FALSE_POSITIVE": nil,
+	}
+	_, valid := validTransitions[fmt.Sprintf("%s->%s", statusPrevious, status)]
+	return valid
```
Comment (jroimartin) [inline pkg/api/service/finding.go:96]: I would do something like this:

```suggestion
	validTransitions := map[string][]string{
		"OPEN":  []string{"FALSE_POSITIVE"},
		"FALSE_POSITIVE":  []string{"OPEN"},
		"FIXED": []string{"FALSE_POSITIVE"},
	}
	return slices.Contains(validTransitions[statusPrevious], status)
```

If we know that there will be always only one valid transition per status, then we can just replace the `map[string][]string` with `map[string]string` and `slices.Contains` with `==`.

**Review by jroimartin — CHANGES_REQUESTED** [review]

File: pkg/api/service/finding.go (hunk 1 above)
Comment (jesusfcr) [reply pkg/api/service/finding.go]: I like it, thanks

**Review by manelmontilla — DISMISSED** [review]
LGTM

File: pkg/api/service/finding.go [hunk 2]
```diff
@@ -87,11 +88,10 @@ func isValidFindingTransition(status, statusPrevious string) bool {
...
 	}
 
-	validTransitions := map[string]any{
-		"OPEN->FALSE_POSITIVE":  nil,
-		"FALSE_POSITIVE->OPEN":  nil,
-		"FIXED->FALSE_POSITIVE": nil,
+	validTransitions := map[string][]string{
+		"OPEN":           []string{"FALSE_POSITIVE"},
+		"FALSE_POSITIVE": []string{"OPEN"},
+		"FIXED":          []string{"FALSE_POSITIVE"},
```
Comment (jroimartin) [inline pkg/api/service/finding.go:94]: You can do what you did in https://github.com/adevinta/vulnerability-db-api/pull/89:

```suggestion
		"OPEN":           {"FALSE_POSITIVE"},
		"FALSE_POSITIVE": {"OPEN"},
		"FIXED":          {"FALSE_POSITIVE"},
```

It looks cleaner.

File: pkg/api/service/finding.go [hunk 3]
```diff
@@ -10,6 +10,7 @@ import (
 
 	"github.com/adevinta/errors"
 	"github.com/adevinta/vulcan-api/pkg/api"
+	"golang.org/x/exp/slices"
```
Comment (jroimartin) [inline pkg/api/service/finding.go:13]: Since Go 1.21 the package `slices` is part of the standard library. Can we update?

**Review by jroimartin — CHANGES_REQUESTED** [review]

File: pkg/api/service/finding.go (hunk 2 above)
Comment (jroimartin) [reply pkg/api/service/finding.go:94]: Have you missed this one or do you prefer to discard the suggestion?

Comment (jesusfcr) [reply pkg/api/service/finding.go]: Sorry I missed, many RPs at the same time :(

**Review by jroimartin — APPROVED** [review]
LGTM. Although the build is failing. Is it related?

**Pull request merged**